src/components/Profile/Profile/FeatureContainer/DoctorsFeature/WorkTimeManagement/WorkTimeManagement.tsx

```diff
@@ -1,66 +1,90 @@
-import { ContainersName, ContainerFooter, ContainerContent } from '../ManagementStyle';
-import {
-  TimeManagementContainer,
-  DateTimePickerWrapper,
-  AddButton,
-} from './WorkTimeManagementStyle';
-
+import { useDispatch } from 'react-redux';
+import { useTypesSelector } from '@hooks/UseTypedSelector';
+import { ERROR_MESSAGE } from '@constants/errorMessage';
+import { profileDoctorAddDate, profileDoctorDelete } from '@store/actionCreators/profileData';
 import TextField from '@mui/material/TextField';
 import AdapterDateFns from '@mui/lab/AdapterDateFns';
 import LocalizationProvider from '@mui/lab/LocalizationProvider';
 import DateTimePicker from '@mui/lab/DateTimePicker';
+import Button from '@components/common/Button/Button';
+import ItemDoctorsFeature from '../ItemDoctorsFeature';
+import { ContainersName, ContainerContent } from '../ManagementStyle';
+import { TimeManagementContainer, DateTimePickerWrapper } from './WorkTimeManagementStyle';
 
 const WorkTimeManagement: React.FC = () => {
+  const dispatch = useDispatch();
   const [date, setDate] = React.useState<Date | null>(null);
-  const [dateItem, setDateItem] = useState()
-  
-  const minTime: Date = new Date('01/01/2021 08:30 AM');
-  const maxTime: Date = new Date('01/01/2021 06:00 PM');
-  
-  const addDate = () => {
-    date ? console.log(date) : console.log('Select time!');
-  };
+  const { workTime } = useTypesSelector((state) => state.profile.profileData);
+  const [addDateError, setAddDateError] = useState<string>('');
+
   const disableWeekends = (date: Date) => {
     return date.getDay() === 0 || date.getDay() === 6;
   };
- 
+
+  const addNewDate = () => {
+    let repeat = false;
+    if (date) {
+      workTime.map((time) => {
+        if (
+          time.time.split(':')[1] ===
+            `${date.getMinutes() < 10 ? '0' + date.getMinutes() : date.getMinutes()}` &&
+          time.date.split('.')[0] === `${date.getDate()}`
+        )
+          repeat = true;
+      });
+      if (repeat) setAddDateError(ERROR_MESSAGE.ADD_DATE_ERROR);
+      else {
+        dispatch(
+          profileDoctorAddDate(
+            `${date.getDate()}.${date.getMonth() + 1}.${date.getFullYear()} ${
+              date.getHours() < 10 ? '0' + date.getHours() : date.getHours()
+            }:${date.getMinutes() < 10 ? '0' + date.getMinutes() : date.getMinutes()}`
+          )
+        );
+        setDate(null);
+        setAddDateError('');
+      }
+    }
+  };
+
+  const deleteDate = (idDate: string) => {
+    dispatch(profileDoctorDelete(idDate));
+  };
+
   return (
     <TimeManagementContainer>
       <ContainersName>Work Time</ContainersName>
 
       <ContainerContent>
-        <ItemDoctorsFeature />
+        {!workTime?.length ? (
+          <span>Have not added time yet</span>
+        ) : (
+          workTime.map((dateTime) => {
+            return <ItemDoctorsFeature key={dateTime._id} deleteDate={deleteDate} {...dateTime} />;
```

Comment:
no need for return statement in arrow function

Reply:
fixed